DS-3889 Tidy RemoveAt.ftable and add tests [revdep skip]

jrwishart opened the pull request:
- Improve removeArrayInputsBad code
- DS-3887 Add method for RemoveAt.ftable
- DS-3889 Add tests
- DS-3889 Add RemoveAt.ftable Rd [ci skip]
- DS-3889 Add ftable rds file for tests
- DS-3889 Improve existing tests [revdep skip]
- DS-3889 Update documentation [ci skip]
- DS-3889 Remove unused code [revdep skip]
- DS-3889 Simplify existing code [revdep skip]

jrwishart committed:
Improve removeArrayInputsBad code

jrwishart committed:
DS-3887 Add method for RemoveAt.ftable

jrwishart committed:
DS-3889 Add tests

jrwishart committed:
DS-3889 Add RemoveAt.ftable Rd [ci skip]

jrwishart committed:
DS-3889 Add ftable rds file for tests

jrwishart committed:
DS-3889 Improve existing tests [revdep skip]

jrwishart committed:
DS-3889 Update documentation [ci skip]

jrwishart committed:
DS-3889 Remove unused code [revdep skip]

jrwishart committed:
DS-3889 Simplify existing code [revdep skip]

jrwishart committed:
Merge remote-tracking branch 'origin/master' into DS-3889-2 [revdep skip]

mwmclean commented on R/removeat.R:
```diff
@@ -80,23 +80,19 @@ RemoveAt.matrix <- function(x, at = NULL, MARGIN = NULL, ignore.case = TRUE, spl
 #' @export
 RemoveAt.ftable <- function(x, at = NULL, MARGIN = NULL, ignore.case = TRUE, split = NULL)
```
Add `@noRd`, `@rdname`, or `@describeIn` so a separate Rd file isn't created.

jrwishart replied:
This applies to all classes for this method as they are also missing it, so I'll do that to all.

mwmclean commented on R/removeat.R:
```diff
@@ -124,21 +134,11 @@ indicesToRetain <- function(names, at, length.x, ignore.case = TRUE, split = NUL
     if (anyNA(at) || !AllIntegers(at))
         stop("'at' must contain character (string) or integer values.")
-    if (any(is.na(at) | at < 1))
+    if (any(at < 0))
```
Why change from `at < 1` to `at < 0`?

jrwishart replied:
Typo, Fixed in [403ae35584b10b692a343945c3b34c18c89c38d1](https://github.com/Displayr/flipU/pull/5/commits/4f12c02931312da47279e8059ef2e2c3db257e84)  thanks

mwmclean commented on tests/testthat/test-removeat.R:
```diff
@@ -198,3 +198,47 @@ test_that("RemoveAt: another vector and a list",
+
+# Check flattened tables are filtered correctly
+
+ftbls <- readRDS("ftbls.rds")
```
If the structure of ftables changes in a later version of R, your tests won't pick that up.

jrwishart replied:
Updated.

jrwishart committed:
DS-3889 Replace QTable with qTable as required [revdep skip]

jrwishart committed:
DS-3889 Remove named args for consistency with other methods [revdep skip]

jrwishart committed:
DS-3889 Fix typo when simplifying code earlier [revdep skip]

jrwishart committed:
DS-3889 Fix documentation [revdep skip]

jrwishart committed:
DS-3889 Update tests as requested [revdep skip]

jrwishart committed:
DS-3889 Support regular ftable [revdep skip]

Added support for standard ftables as well as the previous ftable/qTable
hybrids

jrwishart committed:
fixup! DS-3889 Support regular ftable [revdep skip]

mwmclean commented on R/removeat.R:
```diff
@@ -1,7 +1,7 @@
 #'
 #' @description Removes one or more elements from an object. Dimensions are never
-#' If all elements are removed from a vector, then a 0-length vector is returned.
+#'              If all elements are removed from a vector, then a 0-length vector is returned.
```
Why would you intent this like this? Would be sweet if you could complete the sentence on the previous line or remove it.
